3_summarize.yml

```diff
@@ -85,3 +93,42 @@ targets:
       '3_summarize/src/annual_thermal_metrics.R')
     depends:
       - temp_ranges
+
+##-- Multi-state GLM2 Data Release --##
+
+  glm2_ice_files:
```

Comment:
I don't think you want to strip the hash from the file name info here, do you? That way the `glm2_ice_files` will look to remake like they haven't changed (and not trigger a rebuild of the annual_metrics_ files) even if the hashes have changed, which is an issue. I would suggest taking the yaml in as an input to your `do_annual_metrics_multi_lake()` function.

Reply:
Good point!I did change it in a commit just now. Although, this passes the filenames in to the step that makes the task makefile, so they would only trigger a rebuild of the makefile itself. I thought make/remake could see a filename and figure out the hash to it? There are many instances where we pass a filename around without a hash and remake knows whether or not to rebuild.

Reply:
I'm going to ask _you_ the question in this case:
> I thought make/remake could see a filename and figure out the hash to it? There are many instances where we pass a filename around without a hash and remake knows whether or not to rebuild.

Will it rebuild as intended? If so, why? if not, why?

Reply:
I looked at this earlier and stared blankly at it, then went to a bunch of meetings, and now I am realizing what the key is. The scenarios I am thinking of where we pass in a filename as an argument, are when they were already their own target. A la
```
1_fetch/out/this_is_a_file.rds:
    command: make_a_file()
```
So, really, remake is looking at the target name and deciding to rebuild. In the scenario where the filenames and hashes are stored in a yml, they don't have targets in the pipeline (which is the whole reason we must also include their hash).

Reply:
close, but not quite. Files can be inputs to functions too (not just the target name), and changes to those files would be recognized by remake as a reason to rebuild. But passing an R object of file names as an input will not trigger a rebuild. Reason being, remake doesn't know that R object is a collection of files vs anything else (like a data.frame, list, etc). So it just checks the hash of the _object_, not the hash of the _files_ listed in the object. Since the file names don't change, the hash of the object doesn't change, and no rebuild is triggered.

Reply:
Ok that makes a lot of sense. OK, and as I thought through the chain of events I see where it breaks. 

With the way it was coded before my recent commit, if any of those files changed: the hash in the yml file would update >> which would mean that the yml file has changed and has a new hash >> which would trigger a rebuild of the object target that used `get_filenames_from_ind` >> BUT then the chain of rebuilds would stop here because the filenames haven't updated which means the rebuilt target object of filenames is not any different than it was before.

Reply:
exactly, yes. You got it.